dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectParameterServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+            putMsg(result, Status.PROJECT_PARAMETER_NOT_EXISTS, String.valueOf(code));
+            return result;
+        }
+
+        // check if project parameter name exists
+        ProjectParameter tempProjectParameter = projectParameterMapper.queryByName(projectParameterName);
+        if (tempProjectParameter != null) {
+            log.error("Project parameter name {} already exists", projectParameterName);
+            putMsg(result, Status.PROJECT_PARAMETER_ALREADY_EXISTS, projectParameterName);
+            return result;
+        }
+
+        projectParameter.setParamName(projectParameterName);
+        projectParameter.setParamValue(projectParameterValue);
+
+        if (projectParameterMapper.updateById(projectParameter) > 0) {
+            log.info("Project parameter is updated and id is :{}", projectParameter.getId());
+            result.setData(projectParameter);
+            putMsg(result, Status.SUCCESS);
+        } else {
+            log.error("Project parameter update error, {}.", projectParameterName);
+            putMsg(result, Status.UPDATE_PROJECT_PARAMETER_ERROR);
+        }
+        return result;
+    }
+
+    @Override
+    public Result deleteProjectParametersByCode(User loginUser, long projectCode, long code) {
+        Result result = new Result();
+
+        // check if user have write perm for project
+        Project project = projectMapper.queryByCode(projectCode);
+        boolean hasProjectAndWritePerm = projectService.hasProjectAndWritePerm(loginUser, project, result);
+        if (!hasProjectAndWritePerm) {
+            return result;
+        }
+
+        ProjectParameter projectParameter = projectParameterMapper.queryByCode(code);
+        // check project parameter exists
+        if (projectParameter == null || projectCode != projectParameter.getProjectCode()) {
+            log.error("Project parameter does not exist, code:{}.", code);
+            putMsg(result, Status.PROJECT_PARAMETER_NOT_EXISTS, String.valueOf(code));
+            return result;
+        }
+
+        if (projectParameterMapper.deleteById(projectParameter.getId()) > 0) {
+            log.info("Project parameter is deleted and id is :{}.", projectParameter.getId());
+            result.setData(Boolean.TRUE);
+            putMsg(result, Status.SUCCESS);
+        } else {
+            log.error("Project parameter delete error, {}.", projectParameter.getParamName());
+            putMsg(result, Status.DELETE_PROJECT_PARAMETER_ERROR);
+        }
+        return result;
+    }
+
+    @Override
+    public Result batchDeleteProjectParametersByCodes(User loginUser, long projectCode, String codes) {
+        Result result = new Result();
+
+        if (StringUtils.isEmpty(codes)) {
+            log.error("Project parameter codes is empty, projectCode is {}.", projectCode);
+            putMsg(result, Status.PROJECT_PARAMETER_CODE_EMPTY);
+            return result;
+        }
+
```

Comment:
This here should verify if user has the permission.

Reply:
it is checked below